lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -70,7 +70,11 @@ def request(method, path, params = {}, data = nil, headers = {})
     attempts = 0
     begin
       debug("Making #{method} request to #{uri}")
-      response = oauth_consumer.request(method, uri.to_s, generate_token, {}, data, headers)
+      if [:post, :put, :patch].include?(method)
+        response = oauth_consumer.request(method, uri.to_s, generate_token, {}, data, headers)
+      else
+        response = oauth_consumer.request(method, uri.to_s, generate_token, {}, headers)
```

Comment:
It could be flipped, I did it like that so it's the inverse of what the oauth gem does.

in 0.4.7 it's:
```
      if [:post, :put].include?(http_method)
        data = arguments.shift
      end
```

Later versions added `:patch` so I added it here as well.